src/utils/utils_load_library.h

```diff
@@ -39,22 +39,34 @@ static inline void *util_open_library(const char *filename) {
 static inline int util_close_library(void *handle) {
     // If the FreeLibrary function succeeds, the return value is nonzero.
     // If the FreeLibrary function fails, the return value is zero.
-    return (FreeLibrary(handle) == 0);
+    return (FreeLibrary((HMODULE)handle) == 0);
 }
 
-static inline void *util_get_symbol_addr(void *handle, const char *symbol) {
-    return GetProcAddress(handle, symbol);
+static inline void *util_get_symbol_addr(void *handle, const char *symbol,
+                                         const char *libname) {
+    if (handle == NULL) {
+        if (libname == NULL) {
+            return NULL;
+        }
+        handle = GetModuleHandle(libname);
+    }
+    return GetProcAddress((HMODULE)handle, symbol);
 }
 
 #else /* Linux */
 
 static inline void *util_open_library(const char *filename) {
-    return dlopen(filename, RTLD_LAZY);
+    // NOTE that we use RTLD_GLOBAL which add all loaded symbols to the
+    // global symbol table. These symbols would be used by the Level Zero
+    // provider later
+    return dlopen(filename, RTLD_GLOBAL | RTLD_LAZY);
```

Comment:
I do not like this change :
- How it will affect other libraries which we open - i.e. ttbmaloc. Because we use this function to load other libraries( but you comment say only about level zero. )

It's look like to moved dirty hack from test to library, what is not too great. If you need specific hack for a test it should be implemented at test level, or we need need extra parameter in this function to decide if we need global or not. (i opt for first version here)

Reply:
I propose something like this